include/xtensor/xsort.hpp

```diff
@@ -372,16 +419,17 @@ namespace xt
             result_type result;
             result.resize({de.size()});
 
-            detail::argsort_iter(de.cbegin(), de.cend(), result.begin(), result.end());
+            detail::argsort_iter(de.cbegin(), de.cend(), result.begin(), result.end(), method);
 
             return result;
         }
     }
 
     template <class E>
-    inline auto argsort(const xexpression<E>& e, placeholders::xtuph /*t*/)
+    inline auto
+    argsort(const xexpression<E>& e, placeholders::xtuph /*t*/, sorting_method method = sorting_method::quick)
```

Comment:
I would need help here, I don't quite understand how the placeholder works. Because at the moment, `xt::argsort(arr, xt::sorting_method::stable)` will not work.

Reply:
That is unfortunate. Does your compiler match `sorting_method` with `placeholders::xtuph`?

Reply:
Compilation fails saying:
> error: no matching function for call to 'argsort'

Reply:
Sorry, I did not spot that. It is because there is indeed no matching overload. You would have to have a 
```
argsort(const xexpression<E>& e, sorting_method method)
```
I wonder if you would want it, or if you may just force the user to specify the axis in this case

Reply:
Or we could add this other overload?

Just so I understand, what is the placeholder for?

Reply:
> Or we could add this other overload?

It's just that any further future additions are then even more complicated. It is true that one might not expect those but it is still a thing to always consider

> Just so I understand, what is the placeholder for?

To have e.g. `axes = {0, 2}`.

Reply:
Ha ok, if you specify several axes, it won't call this overload `argsort(const xexpression<E>& e, std::ptrdiff_t axis = -1)` but `rgsort(const xexpression<E>& e, placeholders::xtuph /*t*/)`?

Reply:
Yes. Several axes won't be matching the scalar `std::ptrdiff_t`

Reply:
Thanks @tdegeus.